### PR TITLE
Adds Communication Service

### PR DESCRIPTION
**Pull request by gabru-md**
Adds Communication Service between the Terminal and the Tree Component.

Cheers!

### CODE REVIEW

File: git-visualize/src/app/terminal/terminal.component.ts [hunk 1]
```diff
@@ -20,7 +21,7 @@ export class TerminalComponent implements OnInit {
 
   executeCommand($event){
   	if($event.which === 13){
-  		this.commands.push(this.cli)
+      this.comm.addCommand(this.cli)
```
Comment (yashovardhanagrawal) [inline git-visualize/src/app/terminal/terminal.component.ts]: travis is failing here... 

Linting error 😛

**Comment by yashovardhanagrawal** [comment]
code looks fine to me, just fix travis 😛

**Comment by gabru-md** [comment]
@yashovardhanagrawal the error is something else.
Can someone debug it please in another PR ?
I cannot quite find a fix for it

**Comment by gabru-md** [comment]
@yashovardhanagrawal @ParthS007 @dynamitechetan @geekyd @xeon-zolt review?

File: git-visualize/src/app/tree/tree.component.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#tree{
+	float: right;
+	background-color: purple;
+	color: white;
+	margin-left: -20px;
+	margin-top: 30px;
+	height: 90vh;
+	width: 100%
+}
```
Comment (ParthS007) [inline git-visualize/src/app/tree/tree.component.css]: Add a line at EOF.

File: git-visualize/src/app/tree/tree.component.html [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<p>
-  tree works!
-</p>
+<div id="tree">
+	wow
+</div>
```
Comment (ParthS007) [inline git-visualize/src/app/tree/tree.component.html]: Same

File: git-visualize/src/app/app.module.ts [hunk 4]
```diff
@@ -22,7 +22,10 @@ import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
     FormsModule,
     NgbModule.forRoot()
   ],
-  providers: [],
-  bootstrap: [AppComponent]
+  providers: [CommunicateService],
+  bootstrap: [AppComponent],
+  schemas: [
+    CUSTOM_ELEMENTS_SCHEMA
```
Comment (ParthS007) [inline git-visualize/src/app/app.module.ts]: I think we should avoid capital words. `Custom_Elements_Schema` would be fine.

**Comment by yashovardhanagrawal** [comment]
@gabru-md give me write access for your fork, I will edit this PR a bit

**Comment by yashovardhanagrawal** [comment]
or else you can DIY
first, run `ng lint --fix`
then there will be a few errors remaining that are because it's asking for `space-indentation` and you have done `tab-indentation` just fix them
also, replace `Renderer` with `Renderer2`

first, fix these, then we will see the coverage issue

File: git-visualize/src/app/app.module.ts (hunk 4 above)
Comment (gabru-md) [reply git-visualize/src/app/app.module.ts]: Cannot understand it, sorry.
CUSTOM_ELEMENTS_SCHEMA is a library function

**Comment by gabru-md** [comment]
@ParthS007 merge ?

**Comment by ParthS007** [comment]
Merging it, :+1:

**Pull request merged**